rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbChangeOriginatorNode.java

```diff
@@ -67,7 +80,7 @@ public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNode
 
     @Override
     protected ListenableFuture<List<TbMsg>> transform(TbContext ctx, TbMsg msg) {
-        ListenableFuture<? extends EntityId> newOriginator = getNewOriginator(ctx, msg.getOriginator());
+        ListenableFuture<? extends EntityId> newOriginator = getNewOriginator(ctx, msg.getOriginator(), msg);
```

Comment:
Why do you need to pass the msg.getOriginator() as method argument if the next argument is a msg?